core/arch/arm/kernel/secure_partition.c

```diff
@@ -1809,13 +1808,15 @@ static TEE_Result sp_enter_invoke_cmd(struct ts_session *s,
 	 * as w1 in FFA_INTERRUPT in case of a foreign interrupt.
 	 */
 	rpc_target_info = thread_get_tsd()->rpc_target_info;
-	thread_id = thread_get_id();
-	assert(thread_id <= UINT16_MAX);
+	sp_s->thread_id = thread_get_id();
+	assert(sp_s->thread_id <= UINT16_MAX);
```

Comment:
Is this assert useful? `thread_get_id()` returns a signed short int so this should always be true, but what does this protect us from?